products/compute/api.yaml

```diff
@@ -9871,11 +9871,11 @@ objects:
                           - :TEMPORARY_REDIRECT
                       - !ruby/object:Api::Type::Boolean
                         name: 'stripQuery'
-                        default_value: false
+                        required: true
                         description: |
                           If set to true, any accompanying query portion of the original URL is removed
                           prior to redirecting the request. If set to false, the query portion of the
-                          original URL is retained.
+                          original URL is retained. The default value is false.
```

Comment:
If this is required, it can't have a default

(here and elsewhere)

Reply:
I was going for saying the default so that people who don't want to specify this field but are required to have an idea of what the default normally is. But can remove

Reply:
I don't have a strong opinion, but my guess is it would be more confusing to specify it. @ctavan, what do you think?

Reply:
I just played a bit with the API explorer at https://cloud.google.com/compute/docs/reference/rest/v1/urlMaps/insert

The API does not accept empty `defaultUrlRedirect` objects, I assume the same is true for the `urlRedirect` blocks although I didn't double check.

If I try to insert a new urlMap with these contents:

```json
{
  "name": "empty-url-map",
  "defaultUrlRedirect": {}
}
```

I get an error:

```json
{
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "invalid",
    "message": "Invalid value for field 'resource.defaultUrlRedirect': ''. No fields set for UrlRedirect"
   }
  ],
  "code": 400,
  "message": "Invalid value for field 'resource.defaultUrlRedirect': ''. No fields set for UrlRedirect"
 }
}
```

As soon as I set _any_ field the insert works, e.g.:

```
{
  "name": "empty-url-map",
  "defaultUrlRedirect": {
    "hostRedirect": "example.com"
  }
}
```

I believe instead of making an arbitrary field mandatory I suggest that the _correct_ way of handling this would be to specify

```yaml
at_least_one_of:
  - host_redirect
  - path_redirect
  - prefix_redirect
  - redirect_response_code
  - https_redirect
  - strip_query
```

for all `urlRedirect`/`defaultUrlRedirect` blocks.

This should also be backwards-compatible for people who did specify `strip_query` and it should also be "forward compatible" with those blocks that currently don's specify mandatory stripQuery fields because empty blocks are not accepted by the compute API anyways for these cases anyways.

WDYT?

Reply:
Yeah, that sounds reasonable.

Reply:
Ah, unfortunately because these fields are nested within lists we cannot enforce `at_least_one_of` on them at this point.